### PR TITLE
revised layout

### LINKED ISSUE #657: Bring the sidebar and toc in on wide windows with the default layout
On really wide windows (simulated by zooming out), the sidebar and toc feel too far from the main column of content:

Screenshot 2024-02-02 at 10 43 58 AM

I’d rather we did something like VitePress, where the sidebar and toc are still attached to the main column:

Screenshot 2024-02-02 at 10 44 32 AM

### PR DESCRIPTION
**Pull request by mbostock**
Changes the header to be top-fixed and to extend all the way to the right edge, and moves the table of contents to under the header. Changes the layout on wide windows so that the sidebar and table of contents remain attached to the center column rather than being on the far edges of the window. Moves the header styles to the default theme from our docs config so that the header works “out of the box” without users having to copy these styles from the docs repo (if they chose to have a header).

Before:
Screenshot 2024-09-25 at 9 48 00 AM

After:
Screenshot 2024-09-25 at 9 48 37 AM

Fixes #657.
Supersedes #1023.

### CODE REVIEW

**Comment by mbostock** [comment]
Looks like there’s a bug where the TOC is visible when it should be hidden (because the sidebar is taking priority).

Before:
Screenshot 2024-09-25 at 10 01 06 AM

After:
Screenshot 2024-09-25 at 10 01 11 AM

Also… wondering if there is a quick fix we can do for the table of contents on mobile #1136. Maybe we can just show it inline rather than position-fixed on narrow windows, so that you can quickly jump down from the top to the desired section?

**Comment by mbostock** [comment]
Also… wondering if there is a quick fix we can do for the table of contents on mobile #1136. Maybe we can just show it inline rather than position-fixed on narrow windows, so that you can quickly jump down from the top to the desired section?

File: src/style/layout.css [hunk 1]
```diff
@@ -2,13 +2,36 @@
   --theme-caret: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='16' height='16' viewBox='0 0 16 16'%3E%3Cpath d='M5 7L8.125 9.5L11.25 7' stroke='black' stroke-width='1.5' stroke-linecap='round' fill='none'/%3E%3C/svg%3E");
   --theme-toggle: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='16' height='16' viewBox='0 0 16 16'%3E%3Cpath d='m10.5,11 2.5-3-2.5-3 M6,8h7' fill='none' stroke='black' stroke-width='2'/%3E%3Crect x='2' y='2' fill='currentColor' height='12' rx='0.5' width='2'/%3E%3C/svg%3E");
   --theme-magnifier: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='16' height='16' viewBox='0 0 16 16'%3E%3Cpath stroke='currentColor' stroke-width='2' fill='none' d='M15,15L10.5,10.5a3,3 0 1,0 -6 -6a3,3 0 1,0 6 6'%3E%3C/path%3E%3C/svg%3E");
+  --observablehq-max-width: 1512px;
+  --observablehq-header-height: 2.2rem;
```
Comment (CobusT) [inline src/style/layout.css:6]:
```suggestion
  --observablehq-max-width: 1440px;
```

File: src/style/layout.css [hunk 2]
```diff
@@ -2,13 +2,36 @@
   --theme-caret: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='16' height='16' viewBox='0 0 16 16'%3E%3Cpath d='M5 7L8.125 9.5L11.25 7' stroke='black' stroke-width='1.5' stroke-linecap='round' fill='none'/%3E%3C/svg%3E");
   --theme-toggle: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='16' height='16' viewBox='0 0 16 16'%3E%3Cpath d='m10.5,11 2.5-3-2.5-3 M6,8h7' fill='none' stroke='black' stroke-width='2'/%3E%3Crect x='2' y='2' fill='currentColor' height='12' rx='0.5' width='2'/%3E%3C/svg%3E");
   --theme-magnifier: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='16' height='16' viewBox='0 0 16 16'%3E%3Cpath stroke='currentColor' stroke-width='2' fill='none' d='M15,15L10.5,10.5a3,3 0 1,0 -6 -6a3,3 0 1,0 6 6'%3E%3C/path%3E%3C/svg%3E");
+  --observablehq-max-width: 1512px;
```
Comment (CobusT) [inline src/style/layout.css:5]:
```suggestion
  --observablehq-max-width: 1440px;
```

**Review by CobusT — CHANGES_REQUESTED** [review]

**Review by CobusT — APPROVED** [review]
LGTM! Thanks.

**Pull request merged**